From e3d7d59b2eb1b31258e73da15ab368a388e30e72 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Wed, 6 Dec 2023 17:42:49 +0100
Subject: [PATCH 0809/1141] media: dcmipp: pixeproc: simplify fmt_default
 handling

Cleanup initialization of the subdev wrt default format and limit
to a single structure since src and sink default formats are same
apart from the mbus code.

Change-Id: I279001ea115293a5cf97a209eeb0c3cbb956687c
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/344618
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
---
 .../st/stm32/stm32-dcmipp/dcmipp-pixelproc.c  | 32 +++++++++----------
 1 file changed, 15 insertions(+), 17 deletions(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
index 5de56455db35..88d79c9323c7 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
@@ -220,20 +220,15 @@ struct dcmipp_pixelproc_device {
 	struct v4l2_rect compose;
 };
 
-static const struct v4l2_mbus_framefmt fmt_sink_default = {
+static const struct v4l2_mbus_framefmt fmt_default = {
 	.width = DCMIPP_FMT_WIDTH_DEFAULT,
 	.height = DCMIPP_FMT_HEIGHT_DEFAULT,
 	.code = PIXELPROC_MEDIA_BUS_SINK_FMT_DEFAULT,
 	.field = V4L2_FIELD_NONE,
-	.colorspace = V4L2_COLORSPACE_DEFAULT,
-};
-
-static const struct v4l2_mbus_framefmt fmt_src_default = {
-	.width = DCMIPP_FMT_WIDTH_DEFAULT,
-	.height = DCMIPP_FMT_HEIGHT_DEFAULT,
-	.code = PIXELPROC_MEDIA_BUS_SRC_FMT_DEFAULT,
-	.field = V4L2_FIELD_NONE,
-	.colorspace = V4L2_COLORSPACE_DEFAULT,
+	.colorspace = V4L2_COLORSPACE_REC709,
+	.ycbcr_enc = V4L2_YCBCR_ENC_DEFAULT,
+	.quantization = V4L2_QUANTIZATION_DEFAULT,
+	.xfer_func = V4L2_XFER_FUNC_DEFAULT,
 };
 
 static const struct v4l2_rect crop_min = {
@@ -324,8 +319,8 @@ static void dcmipp_pixelproc_adjust_fmt(struct v4l2_mbus_framefmt *fmt, u32 pad)
 	/* Only accept code in the pix map table */
 	vpix = dcmipp_pixelproc_pix_map_by_code(fmt->code, pad);
 	if (!vpix)
-		fmt->code = IS_SRC(pad) ? fmt_src_default.code :
-					  fmt_sink_default.code;
+		fmt->code = IS_SRC(pad) ? PIXELPROC_MEDIA_BUS_SRC_FMT_DEFAULT :
+					  PIXELPROC_MEDIA_BUS_SINK_FMT_DEFAULT;
 
 	fmt->width = clamp_t(u32, fmt->width, DCMIPP_FRAME_MIN_WIDTH,
 			     DCMIPP_FRAME_MAX_WIDTH) & ~1;
@@ -333,8 +328,7 @@ static void dcmipp_pixelproc_adjust_fmt(struct v4l2_mbus_framefmt *fmt, u32 pad)
 			      DCMIPP_FRAME_MAX_HEIGHT);
 
 	if (fmt->field == V4L2_FIELD_ANY || fmt->field == V4L2_FIELD_ALTERNATE)
-		fmt->field = IS_SRC(pad) ? fmt_src_default.field :
-					   fmt_sink_default.field;
+		fmt->field = V4L2_FIELD_NONE;
 
 	dcmipp_colorimetry_clamp(fmt);
 }
@@ -348,7 +342,9 @@ static int dcmipp_pixelproc_init_cfg(struct v4l2_subdev *sd,
 		struct v4l2_mbus_framefmt *mf;
 
 		mf = v4l2_subdev_get_try_format(sd, state, i);
-		*mf = IS_SRC(i) ? fmt_src_default : fmt_sink_default;
+		*mf = fmt_default;
+		mf->code = IS_SRC(i) ? PIXELPROC_MEDIA_BUS_SRC_FMT_DEFAULT :
+				       PIXELPROC_MEDIA_BUS_SINK_FMT_DEFAULT;
 	}
 
 	return 0;
@@ -852,8 +848,10 @@ dcmipp_pixelproc_comp_bind(struct device *comp, struct device *master,
 	}
 
 	/* Initialize the frame format */
-	pixelproc->sink_fmt = fmt_sink_default;
-	pixelproc->src_fmt = fmt_src_default;
+	pixelproc->sink_fmt = fmt_default;
+	pixelproc->sink_fmt.code = PIXELPROC_MEDIA_BUS_SINK_FMT_DEFAULT;
+	pixelproc->src_fmt = fmt_default;
+	pixelproc->src_fmt.code = PIXELPROC_MEDIA_BUS_SRC_FMT_DEFAULT;
 	pixelproc->crop.top = 0;
 	pixelproc->crop.left = 0;
 	pixelproc->crop.width = DCMIPP_FMT_WIDTH_DEFAULT;
-- 
2.39.2

